### PR TITLE
Fixing ListObjects, ListBucketInfo endpoints

### LINKED ISSUE #62: ListObjects do not work correctly
When I do `mc ls myzcn/buck1/` it behaves weirdly and it does not show the files. Also it does not honor the `-r` flag.

## Current Behavior
1. `mc ls myzcn/` works correctly and shows all the buckets.
2. `mc ls -r myzcn/` does not show all the objects.
3. `mc ls myzcn/buck1/` works correctly and shows a folder named `nest1` at this location.
4. `mc ls -r myzcn/buck1` does not work and shows nothing
5. `mc ls myzcn/buck1/nest1` shows the files at this location whereas `mc ls myzcn/buck1/nest1/` (`/` at the end) does not work and shows nothing.
image

## Expected Behavior
ListObject for `zs3` should work the same as when the host is `s3`.
image

## Steps to Reproduce (for bugs)
1. Deploy `zs3server` locally and use `mc` to do `ls` on various buckets.

### PR DESCRIPTION
**Pull request by sunilmhta**
## Description
Currently the listobjects does not work correctly. Check #62 for all the details.

## How to test this PR?
1. Build and deploy zs3server (this branch)
2. Call mc ls with mc or call the Get endpoint from Postman

### CODE REVIEW

File: cmd/gateway/zcn/dStorage.go [hunk 1]
```diff
@@ -77,43 +77,50 @@ func listRegularRefs(alloc *sdk.Allocation, remotePath, marker, fileType string,
...
-		}
-		if len(oResult.Refs) == 0 {
-			break
-		}
 
-		for i := 0; i < len(oResult.Refs); i++ {
-			ref := oResult.Refs[i]
-			trimmedPath := strings.TrimPrefix(ref.Path, remotePath+"/")
-			if isDelimited {
+	directories := []string{remotePath}
```
Comment (dabasov) [inline cmd/gateway/zcn/dStorage.go:81]: why do we need a loop if there is only one element in the slice?

Comment (sunilmhta) [reply cmd/gateway/zcn/dStorage.go:81]: Initially there is only one element but for the recursive flags i.e `ls -r` we need to find the files inside the nested directories and in those cases this directories slice would be appended. hence the loop.

File: cmd/gateway/zcn/gateway-zcn.go [hunk 2]
```diff
@@ -392,14 +400,8 @@ func (zob *zcnObjects) ListObjects(ctx context.Context, bucket, prefix, marker,
...
 
-	var isSuffix bool
-	if strings.HasSuffix(prefix, "/") {
-		remotePath = filepath.Clean(remotePath) + "/"
-		isSuffix = true
-	}
-
 	var ref *sdk.ORef
-	ref, err = getSingleRegularRef(zob.alloc, remotePath)
+	ref, err = getSingleRegularRef(zob.alloc, filepath.Clean(remotePath))
```
Comment (Hitenjain14) [inline cmd/gateway/zcn/gateway-zcn.go]: Please add filepath.Clean to the function rather than changing the input.

Comment (sunilmhta) [reply cmd/gateway/zcn/gateway-zcn.go]: Done for this as well as other method. Thanks for the suggestion.

**Review by Hitenjain14 — APPROVED** [review]
lgtm

**Pull request merged**